// grunt-php-constants/Gruntfile.js
/*
 * grunt-php-constants
 * https://github.com/davemedema/grunt-php-constants
 *
 * Copyright (c) 2013 Dave Medema
 * Licensed under the MIT license.
 */

'use strict';

module.exports = function(grunt) {

  // Config
  // ---

  grunt.initConfig({

    // package.json
    pkg: grunt.file.readJSON('package.json'),

    // `clean`
    clean: {
      tests: ['tmp']
    },

    // `jshint`
    jshint: {
      options: {
        jshintrc: '.jshintrc'
      },
      all: [
        'Gruntfile.js',
        'tasks/*.js',
        '<%= nodeunit.tests %>'
      ]
    },

    // `nodeunit`
    nodeunit: {
      tests: ['test/*_test.js']
    },

    // `php_constants`
    php_constants: {
      static_option: {
        options: {
          constName: 'TEST',
          constValue: 'static'
        },
        src: 'test/fixtures/test.php',
        dest: 'tmp/static.php'
      },
      template_option: {
        options: {
          constName: 'TEST',
          constValue: '<%= template_test %>'
        },
        src: 'test/fixtures/test.php',
        dest: 'tmp/template.php'
      },
      function_option: {
        options: {
          constName: 'TEST',
          constValue: function() {
            return 'function';
          }
        },
        src: 'test/fixtures/test.php',
        dest: 'tmp/function.php'
      }
    },

    // `template_test`
    template_test: 'template'

  });

  // Load tasks
  // ---

  grunt.loadTasks('tasks');

  // npm tasks
  // ---

  require('load-grunt-tasks')(grunt);

  // Task aliases
  // ---

  grunt.registerTask('default', ['test']);

  grunt.registerTask('release', function(version) {
    grunt.task.run('test');
    grunt.task.run('npm_version:' + (version || 'patch'));
  });
  grunt.registerTask('r', ['release']);

  grunt.registerTask('test', ['clean', 'jshint', 'php_constants', 'nodeunit']);
  grunt.registerTask('t', ['test']);

};
